diff --git a/esm/context.mjs b/esm/context.mjs
index 1d2f4bacbc65053b3a8d415acd52b39e719e9fbd..09b1b0ec2d5f8494c16734353514b0c422b6a180 100644
--- a/esm/context.mjs
+++ b/esm/context.mjs
@@ -9,7 +9,7 @@ const {
   useRef
 } = ReactExports;
 function createContext() {
-  if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production") {
+  if ("production" !== "production") {
     console.warn(
       "[DEPRECATED] `context` will be removed in a future version. Instead use `import { createStore, useStore } from 'zustand'`. See: https://github.com/pmndrs/zustand/discussions/1180."
     );
diff --git a/esm/index.mjs b/esm/index.mjs
index 8b60c00ff0d0cd12da0624353c7a2ae398b80e34..4ba882c09ee8925a017cd83420ea8f8dd22dc523 100644
--- a/esm/index.mjs
+++ b/esm/index.mjs
@@ -8,7 +8,7 @@ const { useSyncExternalStoreWithSelector } = useSyncExternalStoreExports;
 let didWarnAboutEqualityFn = false;
 const identity = (arg) => arg;
 function useStore(api, selector = identity, equalityFn) {
-  if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production" && equalityFn && !didWarnAboutEqualityFn) {
+  if ("production" !== "production" && equalityFn && !didWarnAboutEqualityFn) {
     console.warn(
       "[DEPRECATED] Use `createWithEqualityFn` instead of `create` or use `useStoreWithEqualityFn` instead of `useStore`. They can be imported from 'zustand/traditional'. https://github.com/pmndrs/zustand/discussions/1937"
     );
@@ -25,7 +25,7 @@ function useStore(api, selector = identity, equalityFn) {
   return slice;
 }
 const createImpl = (createState) => {
-  if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production" && typeof createState !== "function") {
+  if ("production" !== "production" && typeof createState !== "function") {
     console.warn(
       "[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`."
     );
@@ -37,7 +37,7 @@ const createImpl = (createState) => {
 };
 const create = (createState) => createState ? createImpl(createState) : createImpl;
 var react = (createState) => {
-  if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production") {
+  if ("production" !== "production") {
     console.warn(
       "[DEPRECATED] Default export is deprecated. Instead use `import { create } from 'zustand'`."
     );
diff --git a/esm/middleware.mjs b/esm/middleware.mjs
index 3304e8d6ea80e5287b659d48b20bdc9a7aeefc23..3d35ff88dd36210f49bb8d15af9596f2c2460072 100644
--- a/esm/middleware.mjs
+++ b/esm/middleware.mjs
@@ -38,11 +38,11 @@ const devtoolsImpl = (fn, devtoolsOptions = {}) => (set, get, api) => {
   const { enabled, anonymousActionType, store, ...options } = devtoolsOptions;
   let extensionConnector;
   try {
-    extensionConnector = (enabled != null ? enabled : (import.meta.env ? import.meta.env.MODE : void 0) !== "production") && window.__REDUX_DEVTOOLS_EXTENSION__;
+    extensionConnector = (enabled != null ? enabled : "production" !== "production") && window.__REDUX_DEVTOOLS_EXTENSION__;
   } catch (_e) {
   }
   if (!extensionConnector) {
-    if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production" && enabled) {
+    if ("production" !== "production" && enabled) {
       console.warn(
         "[zustand devtools middleware] Please install/enable Redux devtools extension"
       );
@@ -95,7 +95,7 @@ const devtoolsImpl = (fn, devtoolsOptions = {}) => (set, get, api) => {
     let didWarnAboutReservedActionType = false;
     const originalDispatch = api.dispatch;
     api.dispatch = (...a) => {
-      if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production" && a[0].type === "__setState" && !didWarnAboutReservedActionType) {
+      if ("production" !== "production" && a[0].type === "__setState" && !didWarnAboutReservedActionType) {
         console.warn(
           '[zustand devtools middleware] "__setState" action type is reserved to set state from the devtools. Avoid using it.'
         );
@@ -569,7 +569,7 @@ const newImpl = (config, baseOptions) => (set, get, api) => {
 };
 const persistImpl = (config, baseOptions) => {
   if ("getStorage" in baseOptions || "serialize" in baseOptions || "deserialize" in baseOptions) {
-    if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production") {
+    if ("production" !== "production") {
       console.warn(
         "[DEPRECATED] `getStorage`, `serialize` and `deserialize` options are deprecated. Use `storage` option instead."
       );
diff --git a/esm/shallow.mjs b/esm/shallow.mjs
index 2457c4296b286d1f33fb6864995b24b6d53103c2..dca0e9b2a13625046481211f0d2c287119a39d34 100644
--- a/esm/shallow.mjs
+++ b/esm/shallow.mjs
@@ -36,7 +36,7 @@ function shallow$1(objA, objB) {
 }
 
 var shallow = (objA, objB) => {
-  if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production") {
+  if ("production" !== "production") {
     console.warn(
       "[DEPRECATED] Default export is deprecated. Instead use `import { shallow } from 'zustand/shallow'`."
     );
diff --git a/esm/vanilla.mjs b/esm/vanilla.mjs
index 3427839341ff0ab88e279054c5c941d0ae8c9f12..0bd60e5445039e77cbea432086c72fafe5e4a89c 100644
--- a/esm/vanilla.mjs
+++ b/esm/vanilla.mjs
@@ -16,7 +16,7 @@ const createStoreImpl = (createState) => {
     return () => listeners.delete(listener);
   };
   const destroy = () => {
-    if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production") {
+    if ("production" !== "production") {
       console.warn(
         "[DEPRECATED] The `destroy` method will be unsupported in a future version. Instead use unsubscribe function returned by subscribe. Everything will be garbage-collected if store is garbage-collected."
       );
@@ -29,7 +29,7 @@ const createStoreImpl = (createState) => {
 };
 const createStore = (createState) => createState ? createStoreImpl(createState) : createStoreImpl;
 var vanilla = (createState) => {
-  if ((import.meta.env ? import.meta.env.MODE : void 0) !== "production") {
+  if ("production" !== "production") {
     console.warn(
       "[DEPRECATED] Default export is deprecated. Instead use import { createStore } from 'zustand/vanilla'."
     );
